src/settings/MappingProfiles/initialDetails/index.js

```diff
@@ -0,0 +1,87 @@
+import ITEM from './ITEM';
+import INSTANCE from './INSTANCE';
+import HOLDINGS from './HOLDINGS';
+import ORDER from './ORDER';
+import INVOICE from './INVOICE';
+import MARC_BIBLIOGRAPHIC from './MARC_BIBLIOGRAPHIC';
+import MARC_HOLDINGS from './MARC_HOLDINGS';
+import MARC_AUTHORITY from './MARC_AUTHORITY';
+
+const values = {
+  ITEM,
+  INSTANCE,
+  HOLDINGS,
+  ORDER,
+  INVOICE,
+  MARC_BIBLIOGRAPHIC,
+  MARC_HOLDINGS,
+  MARC_AUTHORITY,
+};
+
+/**
+ * Retrieves and returns initial mapping details object
+ * for the given entity (record type)
+ *
+ * @param {string} entity record type for which we want to retrieve initial values object
+ * @param {boolean} stripRepeatableFields indicates whether we need strip repeatable fields subfields props
+ * @returns {*}
+ */
+export const getInitialDetails = (entity, stripRepeatableFields = false) => {
+  const current = values[entity];
+
+  if (!stripRepeatableFields) {
+    return current;
+  }
+
+  const fields = current.mappingFields.map(field => ({
+    ...field,
+    subfields: [],
+  }));
+
+  return {
+    ...current,
+    mappingFields: fields,
+  };
+};
+
+/**
+ * Retrieves and returns initial subfields objects for the given entities repeatable fields
+ *
+ * @param {string} entity record type for which we want to retrieve initial values object
+ * @returns {{}}
+ */
+export const getInitialFields = entity => {
+  let initialFields = {};
+  const initialDetails = getInitialDetails(entity);
+  const fields = initialDetails.mappingFields.filter(field => field.subfields && field.subfields.length);
+
+  fields.forEach(field => {
+    initialFields = {
+      ...initialFields,
+      [field.name]: field.subfields[0],
+    };
+  });
+
+  return initialFields;
+};
+
+/**
+ * Retrieves, forms and returns reference tables object from
+ * mapping details fields list/array given
+ *
+ * @param {array} initials
+ * @returns {{}}
+ */
+export const getReferenceTables = initials => {
```

Comment:
It is not clear for me what does "initials" arg mean

Reply:
Done